### PR TITLE
Tma 318 update run tests

### PR DESCRIPTION
**Pull request by jashelb90**
Update the tests to run gradle and send the tests report to the expected directory as provided by the TEST_REPORT_DIR environment variable.

### CODE REVIEW

**Review by Aidan23Smith — APPROVED** [review]
Looks right to me (but I think it will fail the tests for now?)

File: tests/audit-tests/run-tests.sh [hunk 1]
```diff
@@ -5,36 +5,7 @@ set -eu
...
-              "duration": 1
-            }
-          }
-        ]
-      }
-    ]
-  }
-]
-EOF
+./gradlew clean test -PTEST_REPORT_DIR=$TEST_REPORT_DIR
```
Comment (Aidan23Smith) [inline tests/audit-tests/run-tests.sh:9]:
```suggestion
./gradlew clean test
```
As the other comment, we can pass the variable directly in

File: tests/audit-tests/build.gradle [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
+    testImplementation 'org.junit.platform:junit-platform-suite:1.8.2'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.2'
+}
+
+tasks {
+    test {
+        dependsOn assemble, testClasses
+        useJUnitPlatform ()
+        systemProperty("cucumber.execution.parallel.enabled", true)
+        systemProperty("cucumber.plugin", "pretty, json:$TEST_REPORT_DIR/result.json, junit:$TEST_REPORT_DIR/result.xml")
```
[the remark this answers is not in the document]
Comment (jashelb90) [reply tests/audit-tests/build.gradle:30]: Sounds good, a lot cleaner

File: tests/event-processing-tests/build.gradle [hunk 3]
```diff
@@ -29,5 +29,6 @@ tasks {
         useJUnitPlatform ()
         systemProperty("cucumber.execution.parallel.enabled", true)
         systemProperty("cucumber.junit-platform.naming-strategy", "long")
+        systemProperty("cucumber.plugin", "pretty, json:$TEST_REPORT_DIR/result.json, junit:$TEST_REPORT_DIR/result.xml")
```
Comment (jashelb90) [inline tests/event-processing-tests/build.gradle:32]:
```suggestion
        systemProperty("cucumber.plugin", "pretty, json:"+System.getenv("TEST_REPORT_DIR")+"/result.json, junit:"+System.getenv("TEST_REPORT_DIR")+"/result.xml")
```

File: tests/event-processing-tests/run-tests.sh [hunk 4]
```diff
@@ -5,36 +5,7 @@ set -eu
...
-              "duration": 1
-            }
-          }
-        ]
-      }
-    ]
-  }
-]
-EOF
+./gradlew clean test -PTEST_REPORT_DIR=$TEST_REPORT_DIR
```
Comment (jashelb90) [inline tests/event-processing-tests/run-tests.sh:9]:
```suggestion
./gradlew clean test
```

**Review by Aidan23Smith — DISMISSED** [review]
Looks right to me

**Review by Aidan23Smith — APPROVED** [review]
Looks right

**Pull request merged**